### ISSUE TITLE
How to training with EMA in model.train()?

### ISSUE THREAD

**Issue by htwang14** [ultralytics/yolov5#13320]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi, I trained the yolo model using `model.train(data="coco8.yaml", epochs=100, imgsz=640)` and everything worked fine. I suppose the above code is not applying EMA, right? Could you please show me how to apply EMA during training using the `model.train` function?

I've checked your code demo [here](https://github.com/ultralytics/ultralytics/issues/8216#issuecomment-1946401506). But it is not implemented with the `model.train`. Thank you in advance!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @htwang14, thank you for reaching out and for using YOLOv5 🚀! 

To get started on your EMA-related query, I recommend reviewing our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/) which provide comprehensive guides, including information on advanced training methods.

If your inquiry is a 🐛 Bug Report, please share a **minimum reproducible example** to help us diagnose the issue more effectively. 

For custom training questions, such as applying EMA, provide additional details on your setup and confirm you’re following [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

## Requirements

Ensure you have [**Python>=3.8.0**](https://www.python.org/) and all dependencies from [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed, including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). Begin with:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Environments

YOLOv5 can be run in these verified environments with all necessary dependencies:

- **Notebooks**: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>
- **Google Cloud**: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker"></a>

## Status

<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg"></a>

Green badges indicate passing Continuous Integration (CI) tests for YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py), and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) across platforms.

## Introducing YOLOv8 🚀

Be sure to check out the latest [YOLOv8](https://github.com/ultralytics/ultralytics) for enhanced capabilities in object detection, segmentation, and classification. Head to [YOLOv8 Docs](https://docs.ultralytics.com/) and start with:
```bash
pip install ultralytics
```

This is an automated response, but rest assured, an Ultralytics engineer will follow up to assist you shortly 😊.

**Comment by htwang14** [comment]
The issue is more related with the ultralytics repo instead of yolov5. I've opened the same question there and closed this one. Thanks!

**Comment by glenn-jocher** [comment]
@htwang14 thank you for the update. If you need further assistance, feel free to reach out on the Ultralytics repo.
